plugins/outputs/sql/sql.go

```diff
@@ -200,28 +213,130 @@ func (p *SQL) generateInsert(tablename string, columns []string) string {
 		strings.Join(placeholders, ","))
 }
 
-func (p *SQL) tableExists(tableName string) bool {
-	stmt := strings.ReplaceAll(p.TableExistsTemplate, "{TABLE}", quoteIdent(tableName))
+func (p *SQL) fetchColumns(tableName string) error {
+	stmt := p.ColumnQueryTemplate
+	stmt = strings.ReplaceAll(stmt, "{TABLE}", quoteStr(tableName))
+
+	columns, err := p.db.Query(stmt)
+
+	if err != nil {
+		return err
+	}
+
+	defer columns.Close()
+
+	for columns.Next() {
+		var columnName string
+
+		if err := columns.Scan(&columnName); err != nil {
+			return err
+		}
+
+		if !slices.Contains(p.tables[tableName], columnName) {
+			p.tables[tableName] = append(p.tables[tableName], columnName)
+		}
+	}
+
+	return nil
+}
+
+func (p *SQL) createTableIfDoesNotExist(metric telegraf.Metric) error {
+	createStmt := p.generateCreateTable(metric)
+	_, err := p.db.Exec(createStmt)
+	if err != nil {
+		return err
+	}
+
+	err = p.fetchColumns(metric.Name())
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Using implicit declaration and short code is nicer to read as you don't need to worry about the scope. Furthermore, it would be good to add some error prefixes to easier spot the issue when debugging user errors later.
```suggestion
	stmt := p.generateCreateTable(metric)
	if _, err := p.db.Exec(stmt); err != nil {
		return fmt.Errorf("creating table failed: %w", err)
	}

	if err := p.fetchColumns(metric.Name()); err != nil {
		return fmt.Errorf("fetching columns failed: %w", err)
	}

	return nil
```